File: FlutterMockzilla/mockzilla_ios/ios/mockzilla_ios/Sources/mockzilla_ios/ProxyMockzillaLogger.swift
//
//  ProxyMockzillaLogger.swift
//  Pods
//
//  Created by Tom Handcock on 03/06/2025.
//
import mockzilla
import SwiftMockzilla

class ProxyMockzillaLogger: MockzillaLogWriter {
    let flutterApi: MockzillaFlutterApi
    
    init(flutterApi: MockzillaFlutterApi) {
        self.flutterApi = flutterApi
    }
    
    func log(logLevel: MockzillaLogLevel, message: String, tag: String, throwable: KotlinThrowable?) {
        do {
            try flutterApi.log(
                logLevel: BridgeLogLevel.fromNative(logLevel),
                message: message as String,
                tag: tag as String,
                exception: throwable?.message,
                completion: { localResult in
                    // Intentionally blank.
                })
        } catch {
            // Intentionally blank as this is a fire and forget call.
        }
    }
}

File: FlutterMockzilla/mockzilla_android/android/src/main/kotlin/com/apadmi/mockzilla/ProxyMockzillaLogger.kt
package com.apadmi.mockzilla

import BridgeLogLevel
import MockzillaFlutterApi
import android.os.Handler
import android.os.Looper
import com.apadmi.mockzilla.lib.models.MockzillaConfig
import com.apadmi.mockzilla.lib.service.MockzillaLogWriter

open class ProxyMockzillaLogger(
    private val flutterApi: MockzillaFlutterApi,
    private val uiThreadHandler: Handler = Handler(Looper.getMainLooper())
) : MockzillaLogWriter {

    override fun log(
        logLevel: MockzillaConfig.LogLevel,
        message: String,
        tag: String,
        throwable: Throwable?
    ) {
        uiThreadHandler.post {
            flutterApi.log(
                BridgeLogLevel.fromNative(logLevel),
                message,
                tag,
                throwable?.localizedMessage
            ) {
                // Intentionally blank as this call is fire and forget.
            }
        }
    }
}